src/common/components/composite/common/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,98 @@
+import 'react-image-crop/dist/ReactCrop.css';
+import React, { FC, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, Checkbox, Dialog, FormControlLabel, Link } from '@mui/material';
+import AttachFileIcon from '@mui/icons-material/AttachFile';
+import UploadButton from '../../../core/UploadButton';
+import { DialogTitle, DialogActions, DialogContent } from '../../../core/dialog';
+import { useApolloErrorHandler, useConfig, useNotification } from '../../../../../hooks';
+import { useUploadFileMutation } from '../../../../../hooks/generated/graphql';
+import { TranslateWithElements } from '../../../../../domain/shared/i18n/TranslateWithElements';
+import SectionSpacer from '../../../../../domain/shared/components/Section/SectionSpacer';
+
+interface FileUploadProps {
+  onUpload?: (fileCID: string) => void;
+}
+
+const FileUploadButton: FC<FileUploadProps> = ({ onUpload }) => {
+  const { t } = useTranslation();
+  const tLinks = TranslateWithElements(<Link target="_blank" />);
+  const { storage, platform } = useConfig();
+  const notify = useNotification();
+  const handleError = useApolloErrorHandler();
+
+  const [dialogOpened, setDialogOpened] = useState(false);
+  const [confirmation, setConfirmation] = useState(false);
+  const [selectedFile, setSelectedFile] = useState<File>();
+
+  const acceptedFileTypes = useMemo(() => storage?.file.mimeTypes.join(','), [storage]);
+  const MB_LIMIT = storage?.file.maxFileSize ? storage.file.maxFileSize / 1048576 : 0;
+
+  const [uploadFile, { loading }] = useUploadFileMutation({
+    onError: handleError,
+    onCompleted: data => {
+      notify(t('components.file-upload.file-upload-success'), 'success');
+      onUpload?.(data.uploadFile);
+    },
+  });
+
+  const handleSubmit = async () => {
+    if (!selectedFile) return;
+
+    if (storage?.file.maxFileSize && selectedFile.size > storage?.file.maxFileSize) {
+      notify(t('components.file-upload.file-size-error', { limit: MB_LIMIT }), 'error');
+      handleClose();
```

Comment:
Good point. Although we have an error notification, I tried without closing the dialog and it's much better. Otherwise user can miss the notification that something went wrong.